internal/contributebot/design/merge.md

```diff
@@ -0,0 +1,98 @@
+# Contribute Bot Auto-Merge Design
+
+Author: [@zombiezen][]<br>
+Date: 2018-12-17<br>
+Part of [#687][]
+
+[@zombiezen]: https://github.com/zombiezen
+[#687]: https://github.com/google/go-cloud/issues/687
+
+## Objective
+
+As a maintainer of Go Cloud, it is desirable to have pull requests merged in as
+quickly as they can be merged after receiving approvals, assuming there are no
+test breakages or merge conflicts. Currently, merging requires a manual process:
+
+```
+while branch is not up-to-date with master {
+  press update branch button
+  wait for Travis run to finish
+}
+click squash-and-merge
+```
+
+This loop is especially expensive the longer the waiting is, since it increases
+the likelihood that the branch is not up-to-date and thus causing the cycle to
+increase iterations (more waiting!).
+
+The objective of this design is to reduce the amount of maintainer [toil][] in
+this process.
+
+[toil]: https://landing.google.com/sre/sre-book/chapters/eliminating-toil/
+
+## Overview
+
+To reduce maintainer toil, this document proposes a new set of interactions for
+Contribute Bot. If a user writes a comment on a pull request with the text
```

Comment:
anything that includes the substring `/merge`, or only a comment `/^\/merge$/` ?

Reply:
Clarified.